code/game/mecha/mech_bay.dm

```diff
@@ -12,6 +12,9 @@
 	nitrogen = 0.01
 	temperature = TCMB
 
+
+///turf/simulated/floor/mech_bay_recharge_floor/BeforeChange() //I'm told this is a better thing to call this request by. Undefinied proc error though.
+

```

Comment:
`BeforeChange` is defined <a href=https://github.com/ParadiseSS13/Paradise/blob/master/code/game/turfs/turf.dm#L168>here</a>. Make sure you've updated your branch to master by doing `git pull upstream master` first, as this change is rather recent.